src/ast/mod.rs

```diff
@@ -3319,6 +3319,18 @@ pub enum Statement {
         drop_behavior: Option<DropBehavior>,
     },
     /// ```sql
+    /// DROP DOMAIN
+    /// ```
+    /// See [PostgreSQL](https://www.postgresql.org/docs/current/sql-dropdomain.html)
+    ///
+    /// DROP DOMAIN [ IF EXISTS ] name [, ...] [ CASCADE | RESTRICT ]
+    ///
+    DropDomain {
+        if_exists: bool,
+        name: ObjectName,
+        drop_behavior: Option<DropBehavior>,
+    },
```

Comment:
Could we wrap the new statement in a named struct? We're looking to move over to [that pattern](https://github.com/apache/datafusion-sqlparser-rs/issues/1204) for newer statements. e.g.
```rust
struct DropDomain {
    if_exist: bool 
    // ...
}
Statement::DropDomain(DropDomain)
```

Reply:
Oki

Reply:
Done in commit [10dab51](https://github.com/apache/datafusion-sqlparser-rs/pull/1828/commits/10dab5142da31e0fd0768318cc57b389ed6b1de2)